tests/main.test.ts

```diff
@@ -1,86 +1,224 @@
+import { collectLinkedPullRequests } from "../src/handlers/collect-linked-pulls";
 
-beforeAll(() => server.listen());
-afterEach(() => server.resetHandlers());
-afterAll(() => server.close());
+dotenv.config();
+const octokit = jest.requireActual("@octokit/rest");
 
-jest.mock("../src/parser/payload", () => {
-  // Require is needed because mock cannot access elements out of scope
-  const cfg = require("./__mocks__/results/valid-configuration.json");
-  return {
-    stateId: 1,
-    eventName: "issues.assigned",
-    authToken: process.env.GITHUB_TOKEN,
-    ref: "",
-    eventPayload: {
-      issue: { html_url: "https://github.com/ubiquibot/user-activity-watcher/issues/1", number: 1, assignees: [{ login: "ubiquibot" }] },
-      repository: {
-        owner: {
-          login: "ubiquibot",
-        },
-        name: "user-activity-watcher",
-      },
-    },
-    settings: cfg,
-  };
+beforeAll(() => {
+  server.listen();
+});
+afterEach(() => {
+  drop(db);
+  server.resetHandlers();
 });
+afterAll(() => server.close());
 
-describe("Run tests", () => {
-  beforeAll(() => {
-    drop(mockDb);
-    for (const item of dbSeed.issues) {
-      mockDb.issues.create(item);
-    }
+describe("User start/stop", () => {
+  beforeEach(async () => {
+    await setupTests();
   });
 
-  it("Should fail on invalid environment", async () => {
-    const oldEnv = { ...process.env };
-    // @ts-expect-error Testing for invalid env
-    delete process.env.SUPABASE_URL;
-    // @ts-expect-error Testing for invalid env
-    delete process.env.SUPABASE_KEY;
-    await expect(getEnv()).rejects.toEqual(new ValidationException("The environment is" + " invalid."));
-    process.env = oldEnv;
-  });
   it("Should parse thresholds", async () => {
     const settings = Value.Decode(userActivityWatcherSettingsSchema, Value.Default(userActivityWatcherSettingsSchema, cfg));
-    expect(settings).toEqual({ warning: 302400000, disqualification: 604800000 });
+    expect(settings).toEqual({ warning: 302400000, disqualification: 604800000, watch: { optOut: [STRINGS.PRIVATE_REPO_NAME] } });
     expect(() =>
       Value.Decode(
         userActivityWatcherSettingsSchema,
         Value.Default(userActivityWatcherSettingsSchema, {
           warning: "12 foobars",
           disqualification: "2 days",
+          watch: { optOut: [STRINGS.PRIVATE_REPO_NAME] },
         })
       )
     ).toThrow(TransformDecodeError);
   });
   it("Should run", async () => {
-    const result = await run(program, Value.Decode(envConfigSchema, process.env));
-    expect(JSON.parse(result)).toEqual({ status: "ok" });
+    const context = createContext(1, 1);
+    const result = await runPlugin(context);
+    expect(result).toBe(true);
+  });
+
+  it("Should process update for all repos except optOut", async () => {
+    const context = createContext(1, 1);
+    const infoSpy = jest.spyOn(context.logger, "info");
```

Comment:
I'd try to avoid making tests relying on the logger since that can change a lot. It'd be better to spy on functions or check that http mocks have been called

Reply:
I understand where you are coming from with this but the reason why I use the logger in some tests and then only state in others is because I figured that technically the logs are an important feature of our plugins and should probably be covered 

I've written all my test suites this way until now and they are pretty reliant on the logger. Unsure whether you meant this as advice for future unit tests or if I should consider refactoring all of them

---

to confirm, the better approach would be to stub functions which result in no state change to assert they were called and then all others can be confirmed via mock db entries, ignoring logs and confirming input/output essentially?